src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -539,6 +546,15 @@ private void doScheduledSearchForRound (RaptorState outputState) {
                     } else {
                         // We are already on a trip, but check if this stop was reached early enough to back up
                         // to an earlier trip on the same pattern.
+                        // For example, consider two potential boarding stops on a pattern (A and B), with the search
+                        // origin closer to B:
+                        //  A----B
+                        //       |
+                        //       O
+                        // In this setup, the earliestBoardTime at A is less than (before) the earliestBoardTime at B.
+                        // It might be possible to board a certain trip at B, but only a later trip at A. Because A
+                        // is evaluated first (providing updates to downstream stops), we need to check for earlier
```

Comment:
As discussed today, we may want to "re-board" the same trip if it decreases walk distance, created issue #638 for that (separate) idea.